SEMOSS/semoss
# FAISS DB Engine Update: Combining vectors into dataset for a single pkl

## CHANGES

- This updates the FAISS DB engine to create a single master dataset pkl file that includes the vectors in an embedding column. 
- This also combines the vector and dataset file to a single file for each indexed document.
- Previous logic created separate files for vectors and the dataset. 

## Testing
1. Upload a file through a vector model in the vector catalog. 
2. Verify that a single `master_dataset_vectors.pkl` file was created in the default schema directory. (EX: C:\workspace\Semoss\vector\Dev_Testing_FAISS__1222b449-1bc6-4358-9398-1ed828e4f26a\schema\default)
3. Verify that a single file named `{your_file_name}_dataset_vectors.pkl` was created in the indexed files directory (EX: C:\workspace\Semoss\vector\Dev_Testing_FAISS__1222b449-1bc6-4358-9398-1ed828e4f26a\schema\default\indexed_files)
4. Upload another file and verify that there is still just a single `master_dataset_vectors.pkl` file
